packages/core/src/createZoomImageWheel.ts

```diff
@@ -237,87 +237,93 @@ export function createZoomImageWheel(container: HTMLElement, options: ZoomImageW
     }
   }
 
-  // These variables are used for zooming on double tap
-  let touchTimer: NodeJS.Timeout | null = null
-  let startTimestamp = 0
-  let currentValue = 0
-  const endValue = 100
-  let zoomDirection: "in" | "out" = "in"
-  let x = 0
-  let y = 0
-  const durationBetweenTap = 300
+  const animationState = {
+    startTimestamp: null as DOMHighResTimeStamp | null,
+    // the state at the start of the zoom animation
+    start: { x: 0, y: 0, zoom: 0 },
+    // the target state at the end of the zoom animation
+    target: { x: 0, y: 0, zoom: 0 },
+  }
+
+  function animateZoom(touchCoordinate: { x: number; y: number }) {
+    // the `touchCoordinate` should be relative to the container
 
-  function animateZoom(timestamp: number) {
     const currentState = store.getState()
-    const containerRect = container.getBoundingClientRect()
-    const zoomPointX = x - containerRect.left
-    const zoomPointY = y - containerRect.top
-    const isDimensionSwitched = checkDimensionSwitched()
-    const zoomX = isDimensionSwitched ? currentState.currentPositionY : currentState.currentPositionX
-    const zoomY = isDimensionSwitched ? currentState.currentPositionX : currentState.currentPositionY
-    const zoomTargetX = (zoomPointX - zoomX) / currentState.currentZoom
-    const zoomTargetY = (zoomPointY - zoomY) / currentState.currentZoom
 
-    if (!startTimestamp) {
-      startTimestamp = timestamp
-      zoomDirection = currentState.currentZoom > 1 ? "out" : "in"
+    animationState.startTimestamp = null
+    animationState.start = {
+      x: currentState.currentPositionX,
+      y: currentState.currentPositionY,
+      zoom: currentState.currentZoom,
     }
 
-    const progress = timestamp - startTimestamp
-    currentValue = Math.min((progress / finalOptions.dblTapAnimationDuration) * endValue, endValue)
+    if (currentState.currentZoom > 1) {
+      animationState.target = {
+        x: 0,
+        y: 0,
+        zoom: 1,
+      }
+    } else {
+      animationState.target = {
+        zoom: finalOptions.maxZoom,
+        x: touchCoordinate.x * (1 - finalOptions.maxZoom),
+        y: touchCoordinate.y * (1 - finalOptions.maxZoom),
+      }
+    }
 
-    if (zoomDirection === "in") {
-      const newCurrentZoom = clamp(1 + (finalOptions.maxZoom - 1) * (currentValue / 100), 1, finalOptions.maxZoom)
+    function lerp(a: number, b: number, t: number): number {
+      return a * (1 - t) + b * t
+    }
```

Comment:
Oh man, I really miss Math back in university and high school ❤️ What I do nowadays in the industry as FrontEnd engineer has almost nothing to do with numbers, mostly logic 🙈 I should have spent more time on this beautiful stuff 😍